src/AppBundle/Validator/Constraints/IsActivableRestaurantValidator.php

```diff
@@ -29,53 +29,55 @@ public function validate($object, Constraint $constraint)
                 ->addViolation();
         }
 
-        $telephoneErrors = $validator->validate($object->getTelephone(), new Assert\NotBlank());
-        if (count($telephoneErrors) > 0) {
-            $this->context->buildViolation($constraint->telephoneMessage)
-                ->atPath('telephone')
-                ->addViolation();
-        }
-
-        $openingHoursErrors = $validator->validate($object->getOpeningHours(), new Assert\NotBlank());
-        if (count($openingHoursErrors) > 0) {
-            $this->context->buildViolation($constraint->openingHoursMessage)
-                ->atPath('openingHours')
-                ->addViolation();
-        }
-
-        $contractErrors = $validator->validate($object->getContract(), [
-            new Assert\NotBlank(),
-            new Assert\Valid(),
-        ]);
-        if (count($contractErrors) > 0) {
-            $this->context->buildViolation($constraint->contractMessage)
-                ->atPath('contract')
-                ->addViolation();
-        }
-
-        // The validations below only make sense when the restaurant is created
-        if (null !== $object->getId()) {
+        if ($object->getState() !== 'pledge') {
```

Comment:
Ok, we are changing an important piece of code here: we could allow enabling restaurants that are not ready for that (for example, with a missing Stripe account). I think it's the opportunity to **add a unit test** for this `IsActivableRestaurantValidator` class.

Check how it's done in the `OrderValidatorTest` class: you will need to extend `ConstraintValidatorTestCase` to have helper methods to test the validator. 

You should at least test that the validator adds a violation on the "enabled" field, when the restaurant is not in state "pledge".